packages/desktop-client/src/components/transactions/TransactionsTable.js

```diff
@@ -732,47 +735,76 @@ const Transaction = memo(function Transaction(props) {
     setPrevShowZero(showZeroInDeposit);
   }
 
-  function onUpdate(name, value) {
-    if (transaction[name] !== value) {
-      let newTransaction = { ...transaction, [name]: value };
+  var reconciledWarningShowing = false;
 
-      // Don't change the note to an empty string if it's null (since they are both rendered the same)
-      if (name === 'note' && value === '' && transaction.note == null) {
-        return;
-      }
+  function onUpdate(name, value) {
+    // Had some issues with this is called twice which is a problem now that we are showing a warning
+    // modal is the transaction is locked. I added a boolean to guard against showing the modal twice.
+    // I'm still not completely happy with how the cells update pre/post modal. Sometimes you have to
+    // click off of the cell manually after confirming your change post modal for example. The last
+    // row seems to have more issues than others but the combination of tab, return, and clicking out
+    // of the cell all have different implications as well.
 
+    if (transaction[name] !== value) {
       if (
-        name === 'account' &&
-        value &&
-        getAccountsById(accounts)[value].offbudget
+        transaction.reconciled === true &&
+        (name === 'credit' || name === 'debit' || name === 'payee')
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏should we simplify it and expand it to ALL fields? I presume `account` and `date` can also impact the budget. And if we add those.. why not expand it to the last remaining one too: `notes`.

Reply:
I copied YNAB on this one. It's not really about what effects the budget. It's about what effects the reconciled amount. Account, date, and notes shouldn't effect that?

Reply:
If a transaction is moved from account A to B - that would definitely affect the reconciled amount.

Example: I reconcile account A to 100 EUR (because that's what I see in my bank account). I then transfer one of the transactions from account A to account B because I mistakenly had added it to A. Account A now is "reconciled", but no longer does the balance match the reality.

Reply:
@MatissJanis you are correct. I was wrong about the account. YNAB doesn't pop a warning for it but we should. I'll change this.